graphql-java-servlet/src/main/java/graphql/kickstart/servlet/HttpRequestHandlerImpl.java

```diff
@@ -48,37 +51,59 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
 
   private void execute(GraphQLInvocationInput invocationInput, HttpServletRequest request,
       HttpServletResponse response) {
-    try {
-      GraphQLQueryResult queryResult = invoke(invocationInput, request, response);
+    if (request.isAsyncSupported()) {
+      AsyncContext asyncContext = request.startAsync(request, response);
```

Comment:
asyncContext might be pre-existing already. For example, if jetty filter started it already (and set all the timeouts if necessary). In that case it would be great to honor such AsyncContext instead of recreating one.

Reply:
Good point. This can be tested with `isAsyncStarted()`.